### ISSUE TITLE
Alert System support

### ISSUE THREAD

**Issue by sundaramprasad** [grafana-druid-plugin/druidplugin#21]
Hello,
Does Grafana support alerting with druid?

Regards,
Sundaram

**Comment by abhisant** [comment]
The plugin works on Grafana V4. Although I didn't get a chance to test the alerting yet. Will get back to you on this.

**Comment by sundaramprasad** [comment]
I am using the druid plugin and its working fine for graphs and charts. Although I am not able to use alerting. Do let me know of any updates about this.

Thanks and Regards,
Sundaram

**Comment by nikhil2406** [comment]
@sundaramprasad i tried checking this plugin with grafana v4.4.0. I see this error in alert tab. Are you also getting similar error? I'm not sure if this error is due to limitation in plugin or as the error says problem with 'datasource' itself.

<img width="772" alt="screen shot 2017-08-16 at 5 35 58 pm" src="https://user-images.githubusercontent.com/2122777/29391057-7575bcac-82a9-11e7-9de4-e30adf861856.png">

**Comment by sundaramprasad** [comment]
@nikhil2406 : yes, I am also getting the same error. I don't exactly understand what the problem is. Probably plugin doesn't yet support alerting.

**Comment by mhshimul** [comment]
I'm also getting the same error.

**Comment by pparkar** [comment]
Any updates on this? Is this limitation in druid or grafana?

**Comment by abhisant** [comment]
This is the limitation of the plugin. We are trying to find some time in our schedule to implement it, but it will take some time.

**Comment by chinseone** [comment]
I'm working on implementing backend support for Druid alerting in Grafana, some of the queries types are already working. Finishing some enhancement...

**Comment by tutunci** [comment]
Hi
is there any update on this?

Thanks
Maurizio

**Comment by shyamremella** [comment]
@chinseone; were you able to push the backend support for Druid. If its any shape or form, would like to give it a try :)

**Comment by chinseone** [comment]
@tutunci @shyamremella thanks for following up, I've created a PR https://github.com/grafana/grafana/pull/9815, please let me know for any suggestions. Thanks.

**Comment by Gauravshah** [comment]
Needs to wait on https://github.com/grafana/grafana/issues/6841

**Comment by richiesgr** [comment]
Hi
Is there something new about alert in grafana ?
Thanks

**Comment by linxd-cheer** [comment]
Hi, 
today i use plugin of druid ( [repo](https://github.com/GoshPosh/druidplugin)) to implement alert, but i get an error when use 'test rule'
![image](https://user-images.githubusercontent.com/50398651/61199717-850cc500-a711-11e9-9611-8fa373b3b95d.png)

what else operation i need, my env like

grafana 6.5.3
node v8.16.0
npm 6.4.1

i hope get your reply, thx @Gauravshah

**Comment by Gauravshah** [comment]
@linxd-cheer you can try using this branch https://github.com/GoshPosh/druidplugin/tree/druid_alert haven't tested it though

**Comment by linxd-cheer** [comment]
> @linxd-cheer you can try using this branch https://github.com/GoshPosh/druidplugin/tree/druid_alert haven't tested it though

sadly, I download it and move into my grafana plugins directory,  get it again

**Comment by linxd-cheer** [comment]
> @linxd-cheer you can try using this branch https://github.com/GoshPosh/druidplugin/tree/druid_alert haven't tested it though

finally, I found the backend need to modify , I'm use the repo (https://github.com/chinseone/grafana/tree/druid-alert) last 2 commits, but it still need some modify, because it support 4.x.x version, so lower to me (6.1.2)

**Comment by knoguchi** [comment]
Yes, the backend needs to modify.   I can't find the link but the Grafana project already expressed that they don't want to add more datasource backends for alerting.  I don't know why Grafana project designed and implemented the alerts in datasource specific manner.  Anyway, they are working on the [backend plugins](https://github.com/grafana/grafana/issues/6841).    I think it's best to wait for the issue to be closed before adding the alert system support in the druid datasource plugin.

**Comment by Gauravshah** [comment]
@knoguchi this https://github.com/GoshPosh/druidplugin/tree/druid_alert uses the new backend plugins feature

**Comment by stroeovidiu** [comment]
I have managed to use the above plugin but the alerts are not consistent ... the name and value don't match the filters and post aggregators .. any hints in this area ? 
Thanks !

**Comment by Gauravshah** [comment]
@stroeovidiu we have not worked on the backend plugin for sometime now. Not sure about the issues

**Comment by stroeovidiu** [comment]
@Gauravshah : thanks for the swift answer. 
I checked through the code and it seems that the problem is coming from : https://github.com/GoshPosh/druidplugin/blob/druid_alert/datasource.go#L337

The problem is that the metric name that comes into the alerting channel ( slack my case ) is "Druid" -- the hardcoded value at the above line ....

**Comment by Gauravshah** [comment]
@stroeovidiu can you create a PR fixing the issue ? we should be able to merge it in

**Comment by stroeovidiu** [comment]
I will do a PR with all the fixes.
I have discovered that postaggregations on the timeseries is not added to the query .. need to fix that also

**Comment by Gauravshah** [comment]
@stroeovidiu yes, some of the features that were added after alert's initial release are also missing. Like json support etc

**Comment by russki** [comment]
@stroeovidiu any news on the PR? would love to see the basic alerting support 😸

**Comment by junneyang** [comment]
+1
